include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -172,94 +186,51 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
+						constexpr uint8_t maxPlanes = 4;
+						const void* inSourcePixels[maxPlanes] = { inDataAdress, nullptr, nullptr, nullptr };
 
-					return true;
-				};
-
-				auto perOutputRegionEncode = [&](const CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
-				{
-					const auto blockDims = asset::getBlockDimensions(commonExecuteData.outFormat);
-
-					auto encode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
-					{
-						auto localOutPos = readBlockPos * blockDims + commonExecuteData.offsetDifference; // @devshgraphicsprogramming @Criss When it's < 0 it gets weird values so it crashes
-						auto* outDataAdress = commonExecuteData.outData + commonExecuteData.oit->getByteOffset(localOutPos, commonExecuteData.outByteStrides);
-						const auto outDataAdressOffsetScratch = ((readBlockPos.z * state->extent.height + readBlockPos.y) * state->extent.width + readBlockPos.x) * currentChannelCount;
+						for (auto blockY = 0u; blockY < blockDims.y; blockY++)
+							for (auto blockX = 0u; blockX < blockDims.x; blockX++)
+							{
+								decodeType decodeBuffer[maxChannels] = {};
 
-						auto* entryScratchAdress = scratchMemory + outDataAdressOffsetScratch;
-						asset::encodePixelsRuntime(outFormat, outDataAdress, entryScratchAdress); // overrrides texels, so region-overlapping case is fine
+								asset::decodePixelsRuntime(inFormat, inSourcePixels, decodeBuffer, blockX, blockY);
+								memcpy(outDataAdress, decodeBuffer, scratchTexelByteSize);
+							}
 					};
 
-					CBasicImageFilterCommon::executePerRegion(commonExecuteData.outImg, encode, commonExecuteData.outRegions.begin(), commonExecuteData.outRegions.end(), clip);
-
-					return true;
-				};
-
-				bool decodeStatus = commonExecute(state, perOutputRegionDecode);
+					auto& inRegions = state->inImage->getRegions(state->inMipLevel);
+					CBasicImageFilterCommon::executePerRegion(state->inImage, decode, inRegions.begin(), inRegions.end());
+				}
 
-				if (decodeStatus)
 				{
-					auto extent = state->outImage->getMipSize(state->outMipLevel);
-					
 					auto getScratchPixel = [&](core::vector4di32_SIMD readBlockPos) -> decodeType*
 					{
-						const auto offset = ((readBlockPos.z * state->extent.height + readBlockPos.y) * state->extent.width + readBlockPos.x) * currentChannelCount;
-						return scratchMemory + offset;
+						const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(readBlockPos.x, readBlockPos.y, readBlockPos.z, 0), scratchByteStrides);
+						return (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
 					};
 
 					auto sum = [&](core::vectorSIMDi32 readBlockPos) -> void
 					{
-						auto current = getScratchPixel(readBlockPos);
+						decodeType* current = getScratchPixel(readBlockPos);
 
 						auto areAxisSafe = [&]()
```

Comment:
you could have just used `readBlockPos>core::vectorSIMDi32(0,0,0,0)`

Reply:
which gives a boolean vector.

Reply:
corrected in following commit